src/libserialize/serialize.rs

```diff
@@ -911,4 +911,5 @@ impl<T: UseSpecializedDecodable> Decodable for T {
 impl<'a, T: ?Sized + Encodable> UseSpecializedEncodable for &'a T {}
 impl<T: ?Sized + Encodable> UseSpecializedEncodable for Box<T> {}
 impl<T: Decodable> UseSpecializedDecodable for Box<T> {}
+impl<'a, T: ?Sized + Decodable> UseSpecializedDecodable for &'a T {}
```

Comment:
Ah, I see, `?Sized + Decodable` is just `Decodable`, because `Decodable: Sized` (as it returns by-value) so the `?Sized` does nothing.